// @better-auth/infra/dist/crypto-B1NYwfFV.mjs
import { i as INFRA_USER_AGENT } from "./constants-9UwOSy9A.mjs";
import { createFetch } from "@better-fetch/fetch";
//#region src/fetch.ts
function createAPI(options, fetchOptions) {
	return createFetch({
		baseURL: options.apiUrl,
		headers: {
			"user-agent": INFRA_USER_AGENT,
			"x-api-key": options.apiKey
		},
		timeout: options.apiTimeout,
		...fetchOptions
	});
}
function createKV(options, fetchOptions) {
	const headers = { "user-agent": INFRA_USER_AGENT };
	if (options.apiKey) headers["x-api-key"] = options.apiKey;
	return createFetch({
		baseURL: options.kvUrl,
		headers,
		timeout: options.kvTimeout ?? 1e3,
		...fetchOptions
	});
}
//#endregion
//#region src/crypto.ts
function randomBytes(length) {
	const bytes = new Uint8Array(length);
	crypto.getRandomValues(bytes);
	return bytes;
}
function bytesToHex(bytes) {
	return Array.from(bytes).map((b) => b.toString(16).padStart(2, "0")).join("");
}
async function hash(message) {
	const msgBuffer = new TextEncoder().encode(message);
	const hashBuffer = await crypto.subtle.digest("SHA-256", msgBuffer);
	return bytesToHex(new Uint8Array(hashBuffer));
}
/** Keyed HMAC-SHA256 for blinded fingerprints (e.g. failed-login passwords). */
async function hmacSha256Hex(key, message) {
	const encoder = new TextEncoder();
	const cryptoKey = await crypto.subtle.importKey("raw", encoder.encode(key), {
		name: "HMAC",
		hash: "SHA-256"
	}, false, ["sign"]);
	const signature = await crypto.subtle.sign("HMAC", cryptoKey, encoder.encode(message));
	return bytesToHex(new Uint8Array(signature));
}
//#endregion
export { createAPI as a, randomBytes as i, hash as n, createKV as o, hmacSha256Hex as r, bytesToHex as t };
